libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ParsedMediaType.java

```diff
@@ -108,8 +108,9 @@ public static ParsedMediaType parseMediaType(String headerValue) {
 
     public static ParsedMediaType parseMediaType(XContentType requestContentType, Map<String, String> parameters) {
         ParsedMediaType parsedMediaType = requestContentType.toParsedMediaType();
-        parsedMediaType.parameters.putAll(parameters);
-        return parsedMediaType;
+
+        return new ParsedMediaType(parsedMediaType.originalHeaderValue,
+            parsedMediaType.type, parsedMediaType.subType, parameters);
```

Comment:
I dunno how paranoid we oughta be here, but I think this one factory method would still leak mutability. Somebody could pass in a parameters map, and then modify it later, IIUC the `Collections.unmodifiableMap` above only enforces an immutable view, if the underlying map is modified, it doesn't do anything for us.

`this.parameters = Map.copyOf(parameters);` above would protect against this, too, right?

Reply:
Fixed in 6614ef1

Reply:
@joegallo this is great point. you are right -  `unmodifiableMap` is just an immutable view. copyOf is better there. Thanks for the fix!!!